org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -189,6 +190,10 @@ public void updateReplacement(CompletionProposal proposal, CompletionItem item,
 		} else {
 			item.setInsertTextFormat(InsertTextFormat.PlainText);
 		}
+		if (!JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().isCompletionListItemDefaultsInsertTextModeSupport() &&
+			JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().getCompletionItemInsertTextModeDefault() != InsertTextMode.AdjustIndentation) {
+			item.setInsertTextMode(InsertTextMode.AdjustIndentation);
```

Comment:
Is it still necessary to set `AdjustIndentation` to each item given that it's already set in the `itemDefault`?

Reply:
My reading of the logic here is : (client **does not** support `insertTextMode` as a value of `itemDefault` **AND** client has not set the `insertTextMode` (global) default as `AdjustIndentation`).

It would be very strange (maybe even a protocol violation) for a client to claim it doesn't support `insertTextMode` in `itemDefault` while also setting it. If the client did support `insertTextMode` in `itemDefault`, we would have set it to `AdjustIndentation` (based on our logic).

I guess we're saving an extra check by assuming clients will behave nicely. Let me know if this is wrong or not correct to do.

Reply:
In this case I'm setting to `AdjustIndentation` for the cases where the client does not support `InsertTextModeSupport` in `ItemDefaults`

Reply:
Sorry I misunderstood the code here before. LGTM now.